### PR TITLE
Remove last references to ClassMetadataInfo

### PR DESCRIPTION
**Pull request by greg0ire**
It has been deprecated for a long, long time.

`ClassMetadataInfo` itself is left to avoid conflict. It shall be removed right before releasing 3.0.0

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/ClassMetadata.php [hunk 1]
```diff
@@ -4,25 +4,3410 @@
...
+     */
+    public $name;
+
+    /**
+     * READ-ONLY: The namespace the entity class is contained in.
      *
-     * @param string $entityName The name of the entity class the new instance is used for.
-     * @psalm-param class-string<T> $entityName
+     * @var string
+     * @todo Not really needed. Usage could be localized.
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/ClassMetadata.php]: I'm not sure what "Usage could be localized" means.

**Review by SenseException — DISMISSED** [review]

File: lib/Doctrine/ORM/Mapping/ClassMetadata.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/ClassMetadata.php]: :thinking: me neither

**Comment by derrabus** [comment]
This change will be a PITA for future merge-ups. 😓

**Comment by greg0ire** [comment]
@derrabus you mean when touching CMI? Git detected it as a rename, so it should be transparent.

**Comment by derrabus** [comment]
Are you sure? Unless GitHub has a very strage way to display a file rename, it looks like a big code chunk has been moved from one class to another.

**Comment by greg0ire** [comment]
It's a bit confusing: `git show --summary` does not show the rename, yet when I amend the commit, I get this:

```
[remove-cmi 952ccd103] Remove ClassMetadataInfo
 Author: Grégoire Paris <postmaster@greg0ire.fr>
 Date: Sun May 1 23:06:22 2022 +0200
 11 files changed, 28 insertions(+), 65 deletions(-)
 rename lib/Doctrine/ORM/Mapping/{ClassMetadataInfo.php => ClassMetadata.php} (99%)
```

Let me try doing a local merge up.

**Comment by greg0ire** [comment]
When trying a local merge up, there is indeed a conflict, where none of the choices are good (I get a choice between deleted file and modified file, but CM is not modified when I pick "deleted file")

**Comment by greg0ire** [comment]
I tried removing `ClassMetadata`, it helps with `git show --summary` but not with the merge up or with Github

**Comment by greg0ire** [comment]
@derrabus I've postponed the removal of CMI itself, the rest can go now IMO.

**Review by derrabus — APPROVED** [review]

**Pull request merged**